cloudevents/extensions/verifiability.md

```diff
@@ -0,0 +1,443 @@
+These challenges can be addressed through means of adding the necessary information inside the event payloads.
+
+Because the CloudEvents specification [requires](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/spec.md#id) the combination of event `source` and `id` to be unique per event, signature replays for identical events are not considered.
+
+Further, this proposal only aims at *verifiability*.
+It does not aim to enable *confidentiality*.
+Consequently, it does not address the threat of unauthorized parties being able to read CloudEvents that were not meant for them (see [Privacy & Security](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/spec.md#privacy-and-security) in the CloudEvents spec).
+
+While the design in this proposal *can* be used by authorized intermediaries to modify and re-sign events, it explicitly does not aim to provide a cryptographic audit trail of event modifications.
+
+As a general principle, this proposal aims to avoid cryptographic agility in favor of simplicity.
+
+## Constraints
+
+We have set the following constraints for the proposed design:
+
+**Verifiability MUST be OPTIONAL:** This ensures that the additional burden of producing verification material and performing verification only applies when verifiability is desired, which is not always the case.
+
+**The design MUST be backward compatible:** Backward compatibility ensures that producers can produce verifiable events without any knowledge about whether the consumers have been configured to and are able to verify events.
+Consumers that do not support verification can consumer signed events as if they were unsigned.
+
+**The verification material MUST be contained in the same message as the event:** The design aims to be simple and robust, and so the verification material MUST be transported and delivered along with the event that it describes—and not in separate events or even through different channels.
+
+
+## Overview
+
+The producer passes a CloudEvent to the SDK, which creates the verification material and adds it to the CloudEvent. When the consumer’s CloudEvents SDK receives a message with event and verification material, it performs a verification of the signature against the key and passes on a verified event to the consumer:
+
+![An illustration showing how the producer's SDK signs an event, and the consumer's SDK verifies it](verifiability1.png)
+
+The verification material is transported in an [Extension Context Attribute](https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md#extension-context-attributes) called `dssematerial`:
+
+* Type: `String`
+* Description: The [DSSE JSON Envelope](https://github.com/secure-systems-lab/dsse/blob/master/envelope.md) that can be used to verify the authenticity and integrity of CloudEvent.
+* Constraints:
+    * OPTIONAL
+    * If present, MUST be Base64 encoded
+
+The verification material, once Base64 decoded, looks something like this:
```

Comment:
"looks something like" isn't precise enough. Are all of those 5 fields required? If so, then say "MUST adhere to the following form". If the fields can change based on the mechanism used to do the verifying then we'll need to be more creative, so let's discuss because across all mechanisms/formats there MUST be at least one consistent field (payloadType I assume) that people can rely upon to disambiguating them.